### PR TITLE
Feature: help topic organization and primary contact restrictions

### PR DESCRIPTION
**Pull request by KarlPerkins**
Modified from [ Pull Request #3494](https://github.com/osTicket/osTicket/pull/3494) to work with osTicket 1.41.x

This adds support for ability to restrict help topics to particular organizations and primary contacts.

Full details: [Issue #3156](https://github.com/osTicket/osTicket/issues/3156)

### CODE REVIEW

**Comment by spyesx** [comment]
Hi guys :) This feature is much welcomed. Since there is no conflict, I guess it's safe to merge it after 2 months. Please?

**Comment by johnkalangha** [comment]
Hi im using this in my osticket and it really helps me to restrict what my client need to see in helptopic

**Comment by luiizfenix** [comment]
Any way i could use this to upgrade from existing v1.10 to v1.14 directly? I havent upgraded because of the lack of that particular feature. And the really amount of work in order to upgrade with database being so different. So... Do you think this would be possible?

Best Regards,

**Comment by JediKev** [comment]
@luiizfenix 

Your best bet is to upgrade to `1.14.3` ([available on our website](https://osticket.com/download)) and then apply the patch. You'll definitely have issues upgrading to later versions after applying the patch though as you'll be on a signature that will not support upgrading (since the changes are not a part of core).

**TLDR;**
Apply these changes at your own risk as you'll have to do extra work just to be able to upgrade after you apply the changes.

Cheers.

**Comment by luiizfenix** [comment]
Thanks for the quick reply @JediKev, so your advise would be fresh install 1.14.3 with my current database and then apply the patch? or do i need to revert my database back to 1.10 original database and then try to upgrade?

**Comment by JediKev** [comment]
@luiizfenix 

If you have a current osTicket installation of v1.10 then you can simply upgrade to 1.14.3 and then apply the changes listed in this pull request. Be sure to backup the site files and database before you upgrade in case something goes wrong.

Cheers.

**Comment by luiizfenix** [comment]
@JediKev I mean, i currently have v1.10 with the patch from [Pull Request #3494](https://github.com/osTicket/osTicket/pull/3494) So my database is already changed to match this. 
Anyways really thanks for the quick reply. 

Best Regards,

**Comment by JediKev** [comment]
@luiizfenix 

Oh, then you will need to change the signature in the `ost_config` table to the appropriate signature for your version and then you should be able to upgrade the normal way. Keep in mind though that the table mods this pull makes might affect the upgrade (I'm not for sure on this as I have not applied these changes before).

Cheers.

**Comment by JVC-MSF-OCB** [comment]
Is anyone using this on v1.15.4?
@KarlPerkins?
